### ISSUE TITLE
Don't trust client on file size

### ISSUE THREAD

**Issue by jnicklas** [elabs/refile#100]
Currently the client can set the file size via the metadata, we should probably not allow this, instead we should look up the file size in the cache.

**Comment by asiniy** [comment]
It seems we need just to replace these 2 lines, no?
https://github.com/elabs/refile/blob/b0ee465314fd3a92b4588715317d3e795d97ee36/lib/refile/attacher.rb#L86
https://github.com/elabs/refile/blob/b0ee465314fd3a92b4588715317d3e795d97ee36/lib/refile/attacher.rb#L102

**Comment by jnicklas** [comment]
@asiniy no, both of those are fine, it's actually [this one](https://github.com/elabs/refile/blob/b0ee465314fd3a92b4588715317d3e795d97ee36/lib/refile/attacher.rb#L79) which is a bit problematic.

**Comment by jnicklas** [comment]
So I'm a bit torn on this one. On the one hand it's bad to blindly trust the client on this, on the other hand, I can't really think of any situation where this could be exploited for malicious purposes, and it'd definitely add some overhead to have to check the size server-side. When using the S3 storage it's an additional HTTP request. That's not awesome.

**Comment by yellowiscool** [comment]
Hi. I am working on an application with storage quotas, and it would be easier for me to rely on the metadata computed by refile. An additional HTTP request is not a problem in this context, but I agree it should not be the default process.

**Comment by wpiekutowski** [comment]
Each of these 2 ways have pros and cons, depending on a use case. Perhaps we could support both and let devs pick up which one makes the most sense in their case?

**Comment by choonkeat** [comment]
> on the other hand, I can't really think of any situation where this could be exploited for malicious purposes

Content-type sniffing mismatch.

e.g. upload html with malicious script tag with blank/no content type using ".jpg" extension in filename, trick user to access link, user browser find no content-type and attempt to render by sniffing content, renders as html executes script, etc...
